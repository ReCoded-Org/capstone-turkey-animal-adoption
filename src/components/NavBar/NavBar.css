.navlink {
  color: #ff7777 !important;
  font-weight: bold;
  font-size: 1.2rem;
  margin: auto 10px;
}

.bgLight {
  background-color: #fff !important;
}

.language {
  border-radius: 20px !important;
  padding: 0.375rem 1.2rem !important;
  border: 1px solid #ff7777 !important;
  color: #777777;
  margin: 10px 0;
}

.logoHeader {
  margin-left: 5px;
}

@media screen and (min-width: 992px) {
  .language {
    margin: 0 50px 0 10px;
  }

  .logoHeader {
    margin-left: 50px;
  }
}
