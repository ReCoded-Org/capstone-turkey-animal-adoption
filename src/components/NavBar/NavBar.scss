@import "../../style/variable.scss";

.nav-link {
  color: $primary !important;
  font-weight: bold;
  font-size: 1.2rem;
  margin: auto 10px;
}
.bg-light {
  background-color: #fff !important;
}
.language {
  border-radius: $btn-border-radius !important;
  padding: 0.375rem $btn-padding-x !important;
  border: 1px solid $primary !important;
  color: #777777;
  margin: 10px 0;

  @media screen and (min-width: 992px) {
    margin: 0 50px 0 10px;
  }
}
.logoHeader {
  margin-left: 5px;
  @media screen and (min-width: 992px) {
    margin-left: 50px;
  }
}
