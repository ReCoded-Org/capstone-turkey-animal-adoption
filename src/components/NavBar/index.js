import React from "react";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";

function NavBar() {
  return (
    <Navbar bg="dark" variant="dark" expand="lg">
      <Navbar.Brand href="#home">Orders Board</Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="mr-auto">
          <Nav.Link href="/" exact="true">
            Home
          </Nav.Link>
          <Nav.Link href="/about">About</Nav.Link>
          <Nav.Link href="/completed">Completed</Nav.Link>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
}

export default NavBar;
