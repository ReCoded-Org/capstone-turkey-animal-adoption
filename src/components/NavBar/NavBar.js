import React, { useState } from "react";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, Button, Form } from "react-bootstrap";
import logo from "../../images/logo.png";
import Login from "../Login/Login";
import "./NavBar.css";
import SignUpModal from "../SignUpModal/SignUpModal";

function NavBar() {
  const [showLogin, setShowLogin] = useState(false);
  const [showSignUp, setShowSignUp] = useState(false);

  return (
    <>
      <Navbar variant="light" expand="lg" className="bgLight">
        <Navbar.Brand>
          <NavLink to="/" exact>
            <img
              src={logo}
              height="50"
              alt="our Furry Friend"
              className="logoHeader"
            />
          </NavLink>
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="mr-auto">
            <NavLink to="/about" className="nav-link navLink">
              About
            </NavLink>
            <NavLink to="/blog" className="nav-link navLink">
              Blog
            </NavLink>
            <NavLink to="/contact" className="nav-link navLink">
              Contact
            </NavLink>
            <NavLink to="/resources" className="nav-link navLink">
              Resources
            </NavLink>
          </Nav>
          <Button onClick={() => setShowSignUp(true)}>SignUp</Button>
          <Button onClick={() => setShowLogin(true)} className="ml-2">
            login
          </Button>
          <Form.Control as="select" className="w-auto language">
            <option value="English">English</option>
            <option value="Arabic">Arabic</option>
            <option value="Turkish">Turkish</option>
          </Form.Control>
        </Navbar.Collapse>
      </Navbar>
      <SignUpModal show={showSignUp} hideFn={setShowSignUp} />
      <Login show={showLogin} hideFn={setShowLogin} />
    </>
  );
}

export default NavBar;
