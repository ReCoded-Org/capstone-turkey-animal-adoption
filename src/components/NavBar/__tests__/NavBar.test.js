import React from "react";
import renderer from "react-test-renderer";
import NavBar from "../NavBar";

it("NavBar renders correctly", () => {
  const tree = renderer.create(<NavBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
