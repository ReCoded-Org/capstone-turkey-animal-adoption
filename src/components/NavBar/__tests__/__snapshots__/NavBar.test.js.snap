// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<nav
  className="bgLight navbar navbar-expand-lg navbar-light"
>
  <span
    className="navbar-brand"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt="our Furry Friend"
        className="logoHeader"
        height="50"
        src="logo.png"
      />
    </a>
  </span>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        aria-current={null}
        className="nav-link navLink"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <a
        aria-current={null}
        className="nav-link navLink"
        href="/blog"
        onClick={[Function]}
      >
        Blog
      </a>
      <a
        aria-current={null}
        className="nav-link navLink"
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      SignUp
    </button>
    <button
      className="ml-2 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      login
    </button>
    <select
      className="w-auto language form-control"
    >
      <option
        value="English"
      >
        English
      </option>
      <option
        value="Arabic"
      >
        Arabic
      </option>
      <option
        value="Turkish"
      >
        Turkish
      </option>
    </select>
  </div>
</nav>
`;
